kuwala/pipelines/google-poi/src/utils/cat_mapping.py

```diff
@@ -0,0 +1,30 @@
+import os
+import json
+import itertools
+from typing import List
+
+CURDIR = os.path.abspath(os.path.curdir)
+idx=CURDIR.split(os.sep).index('src')
+POI_PATH = (os.sep).join(CURDIR.split(os.sep)[:idx])
+CAT_PATH = os.path.join(POI_PATH, 'resources', 'categories.json')
+
+
+with open(CAT_PATH, 'r') as j:
+    kuwala_to_poi = json.load(j)
+
+def word_contains_x(x:str, word:str)->str:
+    if x in word:
```

Comment:
I think this is part of the reason, you have tags mapped to multiple categories. The tags belonging to a category have to be exact matches. Otherwise, service appears in many Kuwala categories. If you cannot find an exact match, then the unmatched Google tag should be appended to the "misc" tags in the `categories.json` and for the current query, you can also directly return "misc".

Reply:
yes you are exactly right. now it is corrected.